From 77aee963fc4edf433d1b926d5b8ad36d8d9e0c12 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 19 Sep 2014 12:28:36 -0400
Subject: [acpi] Cleanup sparse warning on acpi_os_initialize1()

Message-id: <1411129776-21656-2-git-send-email-prarit@redhat.com>
Patchwork-id: 96339
O-Subject: [RHEL7.1 PATCH BZ 1141846 01/61] ACPI: Cleanup sparse warning on acpi_os_initialize1()
Bugzilla: 1141846
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Myron Stowe <mstowe@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1141846

commit 1129c92faa069581bf3acf34cae92477bd6161d8
Author: Lv Zheng <lv.zheng@intel.com>
Date:   Tue Jul 23 16:11:55 2013 +0800

    ACPI: Cleanup sparse warning on acpi_os_initialize1()

    This patch cleans up the following sparse warning:

    # make C=2 drivers/acpi/osl.o
    ...
    drivers/acpi/osl.c:1775:20: warning: symbol 'acpi_os_initialize1' was not declared. Should it be static?
    ...
      CC      drivers/acpi/osl.o

    Signed-off-by: Lv Zheng <lv.zheng@intel.com>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
Cc: Stefan Assmann <sassmann@redhat.com>
Cc: Tony Camuso <tcamuso@redhat.com>
Cc: Don Zickus <dzickus@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/bus.c b/drivers/acpi/bus.c
index 4a8e373..3efbb17 100644
--- a/drivers/acpi/bus.c
+++ b/drivers/acpi/bus.c
@@ -610,7 +610,6 @@ static int __init acpi_bus_init(void)
 {
  int result;
  acpi_status status;
- extern acpi_status acpi_os_initialize1(void);
 
  acpi_os_initialize1();
 
diff --git a/drivers/acpi/internal.h b/drivers/acpi/internal.h
index a21c08e..f315637 100644
--- a/drivers/acpi/internal.h
+++ b/drivers/acpi/internal.h
@@ -23,6 +23,7 @@
 
 #define PREFIX "ACPI: "
 
+acpi_status acpi_os_initialize1(void);
 int init_acpi_device_notify(void);
 int acpi_scan_init(void);
 void acpi_pci_root_init(void);
diff --git a/drivers/acpi/osl.c b/drivers/acpi/osl.c
index 04c93dd..96d7944 100644
--- a/drivers/acpi/osl.c
+++ b/drivers/acpi/osl.c
@@ -53,6 +53,7 @@
 #include <acpi/acpi.h>
 #include <acpi/acpi_bus.h>
 #include <acpi/processor.h>
+#include "internal.h"
 
 #define _COMPONENT  ACPI_OS_SERVICES
 ACPI_MODULE_NAME("osl");
-- 
1.7.1